### PR TITLE
[feature] Sending emojis in text messages (issue #729)

### PR DESCRIPTION
**Pull request by AdiAr11**
### Types
- [ ] Fix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [x] Refactor (non-breaking change which improves code quality - QA thoroughly )
- [ ] Breaking Change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Chore (updates that would not affect or change the code involving the app itself)

### QA

- [ ] Signup
- [ ] Login
- [ ] Logout
- [ ] Unencrypted Chat 
- [ ] Encrypted Chat 
- [ ] Group Chats
- [ ] Profile Changes
- [ ] Theming Changes
- [ ] Force Kill and Restart

### Final Checklist
 
- [x] All associated issues have been linked to PR
- [ ] All necessary changes made to the documentation
- [ ] Parties interested in these changes have been notified
- [ ] Linter has been run on all code in the PR

### CODE REVIEW

**Comment by AdiAr11** [comment]
Should i resolve the conflicts @ereio

**Comment by AdiAr11** [comment]
Resolved merge conflicts! @EdGeraghty

**Comment by AdiAr11** [comment]
Hi @ereio , I have merged this branch with latest code from dev branch and implemented the Emoji feature again. Please have a look whenever you get time and let me know if changes are required.

File: lib/views/home/chat/widgets/ChatInput.dart [hunk 1]
```diff
@@ -697,7 +706,37 @@ class ChatInput extends HookWidget {
...
           ),
-        )
+        ),
+        Visibility(
+          visible: isEmojiShowing.value,
+          child: SizedBox(
+            height: 250,
+            child: EmojiPicker(
+              textEditingController: controller ,
+              // onEmojiSelected: () => onUpdateInput(controller.value.text),
```
Comment (0x1a8510f2) [inline lib/views/home/chat/widgets/ChatInput.dart]: How come this is commented out?

File: lib/views/home/chat/widgets/ChatInput.dart [hunk 2]
```diff
@@ -697,7 +706,37 @@ class ChatInput extends HookWidget {
...
+                onUpdateInput(controller.value.text);
+              },
+              config: Config(
+                  emojiSizeMax: 28 * (Platform.isIOS ? 1.30 : 1.0),
+                  initCategory: Category.RECENT,
+                  bgColor: Colors.white,
+                  indicatorColor: Colors.blue,
+                  iconColor: Colors.grey,
+                  iconColorSelected: Colors.blue,
+                  // progressIndicatorColor: Colors.blue,
```
Comment (0x1a8510f2) [inline lib/views/home/chat/widgets/ChatInput.dart]: Also this?

**Review by 0x1a8510f2 — DISMISSED** [review]
Looks good code-wise, though I didn't try building. It would be great if you could attach some screenshots. I've left some comments about some commented out code, but otherwise, thanks for working on this!

File: lib/views/home/chat/widgets/ChatInput.dart (hunk 1 above)
Comment (AdiAr11) [reply lib/views/home/chat/widgets/ChatInput.dart]: Sorry, this is not needed. Have written the correct implementation of this just below. Will remove this and the one below it.

**Comment by AdiAr11** [comment]
> Looks good code-wise, though I didn't try building. It would be great if you could attach some screenshots. I've left some comments about some commented out code, but otherwise, thanks for working on this!

Thank you @0x1a8510f2. I will attach a video recording for reference.

**Comment by AdiAr11** [comment]
@0x1a8510f2 Video recording of the implemented feature!!
https://github.com/syphon-org/syphon/assets/54585121/6747fa7c-cdfc-4717-a410-7ffdb2193730

**Comment by ereio** [comment]
Awesome job, was going to ask for a video as well

**Review by ereio — DISMISSED** [review]

**Comment by AdiAr11** [comment]
Thank you @ereio @0x1a8510f2 !!!!

**Review by ereio — APPROVED** [review]

**Pull request merged**